src/Trace/Reporter/ZipkinReporter.php

```diff
@@ -95,98 +95,68 @@ public function report(TracerInterface $tracer)
     }
 
     /**
-     * Convert spans into Zipkin's expected JSON output format.
+     * Convert spans into Zipkin's expected JSON output format. See http://zipkin.io/zipkin-api/#/default/post_spans
+     * for output format.
      *
-     * @param  TracerInterface $tracer
+     * @param TracerInterface $tracer
+     * @param array $headers [optional] HTTP headers to parse. **Defaults to** $_SERVER
      * @return array Representation of the collected trace spans ready for serialization
      */
-    public function convertSpans(TracerInterface $tracer)
+    public function convertSpans(TracerInterface $tracer, $headers = null)
     {
+        $headers = $headers ?: $_SERVER;
         $spans = $tracer->spans();
-        $context = $tracer->context();
-        $traceId = $context->traceId();
+        $rootSpan = $spans[0];
+        $traceId = $tracer->context()->traceId();
+
+        $kindMap = [
+            TraceSpan::SPAN_KIND_CLIENT => 'CLIENT',
+            TraceSpan::SPAN_KIND_SERVER => 'SERVER',
+            TraceSpan::SPAN_KIND_PRODUCER => 'PRODUCER',
+            TraceSpan::SPAN_KIND_CONSUMER => 'CONSUMER'
+        ];
+
+        // True is a request to store this span even if it overrides sampling policy.
+        // This is true when the X-B3-Flags header has a value of 1.
+        $isDebug = array_key_exists('HTTP_X_B3_FLAGS', $headers) && $headers['HTTP_X_B3_FLAGS'] == '1';
 
-        $endpoint = [
+        // True if we are contributing to a span started by another tracer (ex on a different host).
+        $isShared = $rootSpan && $rootSpan->parentSpanId() != null;
```

Comment:
best signal is if you read headers and used those same IDs (as opposed this span being a child of incoming headers). might need a closer look at the code that extracts IDs..